include/aspect/utilities.h

```diff
@@ -1060,6 +1060,13 @@ namespace aspect
      * entry in the list must match one of the allowed operations.
      */
     std::vector<Operator> create_model_operator_list(const std::vector<std::string> &operator_names);
+
+    /**
+     * Create matrix with unit at independent indices
+     */
+    template <int dim>
+    SymmetricTensor<2,dim> symmetric_independent_component_matrix (const unsigned int k);
```

Comment:
For the text, it should be "...WHOSE entries are zero..."

Fundamentally, this function returns the *elements of a basis* for the set of symmetric tensors since every tensor can be written as a linear combination of the tensors returned by this function. So maybe something like `nth_basis_for_symmetric_tensors()`?